test/e2e/tests/request-queuing/chainid-check.spec.js

```diff
@@ -0,0 +1,132 @@
+const { strict: assert } = require('assert');
+const FixtureBuilder = require('../../fixture-builder');
+const {
+  withFixtures,
+  openDapp,
+  unlockWallet,
+  DAPP_URL,
+  regularDelayMs,
+  WINDOW_TITLES,
+  defaultGanacheOptions,
+  switchToNotificationWindow,
+} = require('../../helpers');
+const { PAGES } = require('../../webdriver/driver');
+
+describe('Request-queue chainId proxy sync', function () {
+  it('should preserve per dapp network selections after connecting without refresh calls @no-mmi', async function () {
```

Comment:
this test fails against develop (you can pull down and test it on [this branch](https://github.com/MetaMask/metamask-extension/compare/chainid-check?expand=1)) because the proxy doesn't update correctly without a refresh prior to the changes in this PR